pallets/rc-migrator/src/preimage.md

```diff
@@ -0,0 +1,191 @@
+The preimage pallet consists of three storage maps, one of which is a legacy item. The interesting one are `RequestStatusFor` and `PreimageFor`.
+
+## Storage: PreimageFor
+
+[Maps](https://github.com/paritytech/polkadot-sdk/blob/00946b10ab18331f959f5cbced7c433b6132b1cb/substrate/frame/preimage/src/lib.rs#L185) a hash and the length of a preimage to its preimage data. Preimages can be migrated rather easily by sending them in chunks from the Relay and appending them on the Asset Hub. The preimages are often used to store large governance calls.
+
+Q: One question here would be whether or not we want to translate these calls. I think we can and should. But I am not sure about the best time point to do so. We can translate the preimages calls upon arrival on the Asset Hub, although there is a small change that a preimage that was not intended to be decoded as a call would be translated. After all, the preimage pallet is a universal data storage without any implied semantics about its content. It could therefore be better to translate the preimages once we are migrating the Referenda pallet and then only translate all preimages that occur as hash lookups in a referenda.
```

Comment:
only scheduler which used by referenda or root can request a preimage. in theory there should only preimages with valid calls having request count > 0.

Reply:
Oh right, so we could also do the translation as we migrate the preimages.

But then when we do the governance migration afterwards it will not find the hashes anymore. So they kind of need to happen together anyway.

Reply:
Right. Might be worth migrating them together